sys/foraging.go

```diff
@@ -376,13 +394,23 @@ func (s *Foraging) searching(w *ecs.World) {
 	}
 
 	s.patches = s.patches[:0]
+	s.foragershuffle = s.foragershuffle[:0]
 }
 
 func (s *Foraging) collecting(w *ecs.World) {
 	sz := float64(s.foragerParams.SquadronSize)
-	foragerQuery := s.foragerFilterLoad.Query()
-	for foragerQuery.Next() {
-		act, patch, milage, load := foragerQuery.Get()
+	activityQuery := s.activityFilter.Query()
+	for activityQuery.Next() {
+		act := activityQuery.Get()
+		if act.Current == activity.Experienced || act.Current == activity.BringPollen || act.Current == activity.BringNectar {
```

Comment:
In the loop over the shuffled entities, we only handle `Experienced`. So why do we need the others too here?